dataRetriever/factory/shard/resolversContainerFactory.go

```diff
@@ -515,19 +517,17 @@ func (rcf *resolversContainerFactory) generateTrieNodesResolver() ([]string, []d
 	keys := make([]string, 0)
 	resolverSlice := make([]dataRetriever.Resolver, 0)
 
-	identifierTrieNodes := factory.TrieNodesTopic + shardC.CommunicationIdentifier(shardC.SelfId())
-
-	resolver, err := rcf.createTrieNodesResolver(identifierTrieNodes)
+	identifierTrieNodes := factory.AccountTrieNodesTopic + shardC.CommunicationIdentifier(shardC.SelfId())
+	resolver, err := rcf.createTrieNodesResolver(identifierTrieNodes, factoryTries.UserAccountTrie)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	resolverSlice = append(resolverSlice, resolver)
 	keys = append(keys, identifierTrieNodes)
 
-	identifierTrieNodes = factory.TrieNodesTopic + shardC.CommunicationIdentifier(sharding.MetachainShardId)
-
-	resolver, err = rcf.createTrieNodesResolver(identifierTrieNodes)
+	identifierTrieNodes = factory.ValidatorTrieNodesTopic + shardC.CommunicationIdentifier(shardC.SelfId())
+	resolver, err = rcf.createTrieNodesResolver(identifierTrieNodes, factoryTries.UserAccountTrie)
```

Comment:
`PeerAccountTrie` instead of `UserAccountTrie`.

Reply:
done.